spec/controllers/spree/products_controller_spec.rb

```diff
@@ -8,7 +8,7 @@
     let!(:store) { FactoryGirl.create(:store) }
 
     it 'should return 404' do
-      spree_get :show, :id => product.to_param
+      spree_get :show, id: product.to_param, format: :json
```

Comment:
@priyank-gupta Why only for format: :json?

Maye we should add spec for html too? It should [raise 404](https://github.com/spree-contrib/spree-multi-domain/blob/master/app/controllers/spree/products_controller_decorator.rb#L10), right?